parser.py

```diff
@@ -1,17 +1,30 @@
 #!/bin/env python
 import argparse
 import pprint
+import re
+
 import grammar
 import visitor
 
+class PreProcessor:
+    @staticmethod
+    def remove_comments(program: str):
+        lines = program.splitlines()
+        cleaned_lines = []
+        for line in lines:
+            cleaned_line = re.sub(r'#.*$', '', line).strip()
+            if cleaned_line:
+                cleaned_lines.append(cleaned_line)
+        return "\n".join(cleaned_lines)
 
 class ImperivmParser:
     def __init__(self, grammar=grammar.imperivm, visitor=visitor.ImperivmVisitor()):
```

Comment:
Example with dep. injection.

```suggestion
    def __init__(self, grammar=grammar.imperivm, visitor=visitor.ImperivmVisitor(), preprocessor=PreProcessor()):
```